packages/thirdweb/src/pay/utils/definitions.ts

```diff
@@ -1,64 +1,69 @@
 import { getThirdwebDomains } from "../../utils/domains.js";
 
+const PAY_DOMAIN: string = getThirdwebDomains().pay;
+const PAY_BASE_URL: string = PAY_DOMAIN.startsWith('localhost')
+   ? `http://${PAY_DOMAIN}`
+   : `https://${PAY_DOMAIN}`;
+
 /**
  * Endpoint to get the status of a "Buy with Crypto" quote.
  * @internal
  */
 export const getPayBuyWithCryptoStatusUrl = () =>
-  `https://${getThirdwebDomains().pay}/buy-with-crypto/status/v1`;
+  `${PAY_BASE_URL}/buy-with-crypto/status/v1`;
```

Comment:
make `PAY_BASE_URL` a function and calculate it when the function it is used in is called